api/restapi/restapi.go

```diff
@@ -77,7 +92,30 @@ func NewRESTAPI(apiMAddr ma.Multiaddr) (*RESTAPI, error) {
 	if err != nil {
 		return nil, err
 	}
+	return newRESTAPI(apiMAddr, l)
+}
+
+// NewTlsRESTAPI creates a new REST API component that uses TLS for security
+// (authentication, encryption). It receives the multiaddress on which the API
+// listens, as well as paths to certificate and private key files
+func NewTLSRESTAPI(apiMAddr ma.Multiaddr, certFile, keyFile string) (*RESTAPI, error) {
```

Comment:
Let this function receive a tls.Config directly: cluster integrator's responsability to set it up.

For ipfs-cluster-service, we are the integrator, so you have to set it up there, :)